### PR TITLE
[Mosaic GPU] Query SM and PTX ISA dynamically using driver and LLVM

### PR DESCRIPTION
**Pull request by andportnoy**
Use CUDA driver API to query major and minor compute capabilities, thus arriving at a "base" SM string (e.g. `sm_90`).
Then use LLVM to see if we can "upgrade" the base SM string to one that enables architecture-specific capabilities (e.g. `sm_90a`).
Then use LLVM to map the SM string to a PTX ISA version that supports the SM.

### CODE REVIEW

File: jaxlib/mosaic/gpu/custom_call.cc [hunk 1]
```diff
@@ -89,8 +92,82 @@ namespace {
...
+  if (subtarget_info == nullptr) {
+    return absl::InternalError(
+        absl::StrFormat("Failed to get LLVM subtarget info for sm %s", sm));
+  }
+
+  for (const llvm::SubtargetFeatureKV& feature :
+       subtarget_info->getEnabledProcessorFeatures()) {
+    if (absl::StartsWith(feature.Key, "ptx")) {
+      std::string ptx_isa = feature.Key;
+      return std::pair<std::string, std::string>(sm, ptx_isa);
```
Comment (apaszke) [inline jaxlib/mosaic/gpu/custom_call.cc:161]: you should be able to `std::make_pair` to avoid the extra annotations I think

**Review by apaszke — APPROVED** [review]
If XLA has utilities for this then I think it would be fine to use those as well. But this LGTM

File: jaxlib/mosaic/gpu/custom_call.cc (hunk 1 above)
Comment (andportnoy) [reply jaxlib/mosaic/gpu/custom_call.cc]: Thank you, fixed in the latest push.

**Comment by andportnoy** [comment]
Superseded by #24449.

**Pull request closed without merging**